CISLibrary/drivers/HBridge/DRV8837.cpp

```diff
@@ -60,31 +47,22 @@ mbed_error_status_t DRV8837::set(double value) {
  * @brief Stop the output, brake or coast
  *
  * @param brake will short output if true, high impedance if false
- * @param blocking will wait until operation is complete if true
+ * @param blocking parameter ignored on the DRV8837 implementation
  * @return mbed_error_status_t
  */
-mbed_error_status_t DRV8837::stop(bool brake) {
-  this->wake();
-  this->rev = this->fwd = (brake ? 1.0 : 0.0);
-
-  wait_us(DRV8837_SET_WAIT_TIME_US);
-  this->sleep_immediate();
+mbed_error_status_t DRV8837::stop(bool brake, bool blocking) {
+  this->setSleep(false);
+  this->rev = this->fwd = (brake ? 1.0f : 0.0f);
+  this->setSleep(true);
 
   return MBED_SUCCESS;
 }
 
 /**
- * @brief Wakes the Hbridge.
- * 
- */
-void DRV8837::wake() {
-  this->nSleep = 0;
-}
-
-/**
- * @brief Force the Hbridge to sleep.
- * 
+ * @brief Sets the HBridge operating mode
+ * @param sleep sets the operating mode to sleep if true, and awakens the
+ * HBridge is false.
```

Comment:
The sleep signal is not to wake up logic to change the PWM then sleep again. The DS has a poor name. It is an enable output signal. When high the output is given by the input: for, rev, brake, coast. When nSleep is low the output is coasting and ignoring any of the PWM input and enters a low power state called sleeping.